sdk/storage/azure-storage-files-datalake/src/datalake_directory_client.cpp

```diff
@@ -125,18 +125,20 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
   FileClient DirectoryClient::GetFileClient(const std::string& path) const
   {
     auto builder = m_dfsUri;
-    builder.AppendPath(path, true);
+    builder.AppendPath(path);
     auto blobClient = m_blobClient;
-    blobClient.m_blobUrl.AppendPath(path, true);
-    return FileClient(std::move(builder), std::move(blobClient), m_pipeline);
+    blobClient.m_blobUrl.AppendPath(path);
+    auto blockBlobClient = blobClient.GetBlockBlobClient();
+    return FileClient(
+        std::move(builder), std::move(blobClient), std::move(blockBlobClient), m_pipeline);
```

Comment:
For the record, I don't think it's right for `FileClient` to include two blob clients. But I can live with it as long as it works.